lib/std/child_process.zig

```diff
@@ -699,6 +714,13 @@ pub const ChildProcess = struct {
             StdIo.Close => {
                 g_hChildStd_IN_Rd = null;
             },
+            StdIo.File => {
+                if (self.stdin) |stdin| {
+                    g_hChildStd_IN_Rd = try dupWinHandleInherit(stdin.handle);
```

Comment:
This should only transfer handles that are marked as inheritable, but we can limit which ones, specifically, are inherited through `UpdateProcThreadAttribute`, where we pass in which handles we want to let the child inherit (which would include STD(IN, OUT, ERR)) if this is desirable. Let me know what seems like the best idea!

Reply:
Uh, that sounds interesting. I think that I did not try that yet. Would this mean that the number of syscalls would or could increase and in what cases?

Reply:
(Forgive me if some of these comments overlap a bit) It would increase the number of syscalls, but if done properly it should only be 4 total for any mixture of passing STD(IN, OUT, ERR). This is because
1. Call `InitializeProcThreadAttributeList` with `NULL`, number of handles to pass, 0, and &u16 to get the size of the buffer.
2. Call `InitializeProcThreadAttributeList` with the previous value to create the list
3. Call `UpdateProcThreadAttribute` with attribute `PROC_THREAD_ATTRIBUTE_HANDLE_LIST` 
4. Call `DeleteProcThreadAttributeList` on the list after spawning.
I think that's how it should be done, but I'd have to test. I've been piecing this together from the Win32 docs

Reply:
Thanks, if you are curious: I did a Windows implementation [here](https://github.com/matu3ba/sandboxamples/blob/master/test/win/main_explicit_handle_inheritance.zig#L63).